### PR TITLE
`gatsby-source-filesystem` Check if file exists locally before downloading

### PR DESCRIPTION
**Pull request by awesomebob**
See #4464

### CODE REVIEW

**Comment by awesomebob** [comment]
@pieh, can you think of anything this PR needs before merging?

**Comment by pieh** [comment]
@awesomebob This looks fine for me!

File: packages/gatsby-source-filesystem/src/create-remote-file-node.js [hunk 1]
```diff
@@ -100,15 +117,7 @@ module.exports = ({ url, store, cache, createNode, auth = {} }) => {
...
-      // Create the file node and return.
-      createFileNode(filename, {}).then(fileNode => {
-        // Override the default plugin as gatsby-source-filesystem needs to
-        // be the owner of File nodes or there'll be conflicts if any other
-        // File nodes are created through normal usages of
-        // gatsby-source-filesystem.
-        createNode(fileNode, { name: `gatsby-source-filesystem` })
-        resolve(fileNode)
-      })
+      createFileSystemNode()
```
Comment (KyleAMathews) [inline packages/gatsby-source-filesystem/src/create-remote-file-node.js]: Could you not create a new function please

Comment (awesomebob) [reply packages/gatsby-source-filesystem/src/create-remote-file-node.js]: I can change it, how would you prefer it to be handled?
The code needs to `createFileNode` whether or not it has to download the remote file, and originally that function was being called on `responseStream.on(`end)'.
Would you rather I just write out the function twice?

Comment (KyleAMathews) [reply packages/gatsby-source-filesystem/src/create-remote-file-node.js]: Hmmm I'm not sure what this PR is fixing now. Files aren't redownloaded — a 304 response has an empty body.

**Comment by pieh** [comment]
Can we close this PR as it turns out this doesn't actually do anything?

**Comment by awesomebob** [comment]
@pieh Yes, I think we can.

**Pull request closed without merging**